scripts/attachSession.js

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env node
+
+'use strict';
+
+var http = require('http'),
+    spawn = require('child_process').spawnSync;
+
+var sessionId = '';
+
+// 1. Create a new selenium session.
+var postData = JSON.stringify({'desiredCapabilities': {'browserName': 'firefox'}});
+var createOptions = {
+  hostname: 'localhost',
+  port: 4444,
+  path: '/wd/hub/session',
+  method: 'POST',
+  headers: {
+    'Content-Type': 'application/x-www-form-urlencoded',
+    'Content-Length': Buffer.byteLength(postData)
+  }
+};
+var req = http.request(createOptions, function(res) {
+  res.on('data', setBody);
+  res.on('end', checkSession);
+});
+req.write(postData);
+req.end();
+
+// 2. After making the request to create a selenium session, read the selenium
+// session id.
+var setBody = function(chunk) {
+  var body = chunk.toString();
+  sessionId = JSON.parse(body).sessionId;
+};
+
+// 3. After getting the session id, verify that the selenium session exists.
+// If the session exists, run the protractor test.
+var checkSession = function() {
+  var checkOptions = {
+    hostname: 'localhost',
+    port: 4444,
+    path: '/wd/hub/session/' + sessionId,
+    method: 'GET'
+  };
+  var state = '';
+  var req = http.request(checkOptions, function(res) {
+    res.on('data', function(chunk) {
+      state = JSON.parse(chunk.toString()).state;
+    });
+    res.on('end', function() {
+      if (state === 'success') {
+        var runProtractor = spawn('bin/protractor',
+            ['spec/attachSession.js', '--seleniumSessionId=' + sessionId]);
```

Comment:
Can we check on the `ChildProcess#close` event and check the exit value of the Protractor process? Then we can also ensure that protractor finishes before stopping the session.